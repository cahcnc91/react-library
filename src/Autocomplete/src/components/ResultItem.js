import React from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  nonSelected: {
    cursor: "pointer",
    display: "flex",
    padding: ".5rem 1.1rem"
  },
  selected: {
    cursor: "pointer",
    display: "flex",
    padding: ".5rem 1.1rem",
    backgroundColor: "rgba(0, 0, 0, .06)"
  },
  text: {
    fontFamily: 'Helvetica, sans-serif',
  },
});

/** displays search items in search drop down */
const ResultItem = ({
  item,
  index,
  activeOption,
  changeActiveOption,
  handleClickItem
}) => {
  const classes = useStyles();

  return (
    <div
      className={
        activeOption === index ? classes.selected : classes.nonSelected
      }
      onClick={() => handleClickItem(item)}
      onMouseOver={() => changeActiveOption(index)}
    >
      {item.label}
    </div>
  );
};

export default ResultItem;
