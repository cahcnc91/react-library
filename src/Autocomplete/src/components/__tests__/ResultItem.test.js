import React from "react";
import { shallow } from "enzyme";
import ResultItem from "../ResultItem";

const handleClickItem = () => {};

const changeActiveOption = () => {};

const item = { label: "JavaScript", id: 1 };
describe("ResultItem", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <ResultItem
        item={item}
        index={0}
        activeOption={1}
        handleClickItem={handleClickItem}
        changeActiveOption={changeActiveOption}
      />
    );
  });

  test("snapshot renders", () => {
    let tree = wrapper.html();
    expect(tree).toMatchSnapshot();
  });

  it("renders the inner text", () => {
    expect(wrapper.text()).toEqual("JavaScript");
    expect(wrapper.find(`[data-test='result-item-1']`))
  });
});
