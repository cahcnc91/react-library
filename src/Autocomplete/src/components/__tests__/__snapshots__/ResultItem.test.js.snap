// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem snapshot renders 1`] = `"<div class=\\"nonSelected-0-2-1\\" data-test=\\"result-item-1\\">JavaScript</div>"`;
